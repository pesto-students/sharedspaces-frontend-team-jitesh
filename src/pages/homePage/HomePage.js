import React, { useEffect, useState } from 'react'
import { useSelector, useDispatch } from 'react-redux'
import './homePage.scss'
import SearchInput from "../../components/searchInput/SearchInput";
import Button from "../../components/button/Button";
import Footer from "../../components/footer/Footer";
import { Link, useNavigate } from "react-router-dom";
import { getAllProperty } from '../../store/actions/siteAction'
import Loader from '../../components/loader/Loader'

const HomePage = () => {
  const dispatch = useDispatch()
  const navigate = useNavigate()
  const [loading, setLoading] = useState(false)
  const [isUserLoggedIn, setIsUserLoggedIn] = useState(false)

  const allProperties = useSelector(state => state.site.allProperties)
  const userDetail = useSelector(state => state.site.userDetail)


  useEffect(() => {
    let data = {}
    dispatch(getAllProperty(data, (value) => setLoading(value)))
  }, [])

  useEffect(() => {
    if (userDetail) {
      setIsUserLoggedIn(true)
    }
  }, [userDetail, userDetail?.token, userDetail?.role])


  const onListYourSpace = () => {
    if (isUserLoggedIn) {
      if (userDetail.role === "User") {
        navigate('/list-your-space')
      } else {
        navigate('/admin/property')
      }
    } else {
      navigate('/list-your-space')
    }
  }


  return (
    <>
      <div className="homepage-wrapper">
        <div className="search-wrapper">
          <div className="search-bg flex">
            <div className="left-bg bg-black">
            </div>
            <div className="right-bg bg-red-500">
            </div>
          </div>
          <div className="container-wrapper">
            <div className="section fade-in-bottom">
              <div className="left-section">
                <div className="search-with-heading">
                  <h1 className="text-white text-5xl font-bold py-10 leading-tight">
                    The  Best Place <br />
                    To Find Your<br />
                    Office Space
                  </h1>
                  <SearchInput />
                </div>
              </div>
              <div className="right-section">
                <img src="/assets/images/search-section.png" alt="" />
              </div>
            </div>
          </div>
        </div>

        <div className="black-background">
          <div className="property-list-wrapper">
            <div className="popular-properties-section">
              <h1 className="text-black text-2xl font-bold py-10 leading-tight  ">
                Popular Properties
              </h1>

              {loading ?
                <div className="flex w-100 justify-center mt-10">
                  <Loader width={"w-10"} className={"text-gray-200"} />
                </div> :

                <div className="property-list">
                  {allProperties?.slice(0, 3)?.map(property =>
                    <Link to={`/property/${property._id}`} className="property-item bg-white shadow-new fade-in-bottom">
                      <div className="property-image">
                        <img src={property.propertyImage} alt={property.propertyTitle} />
                      </div>
                      <div className="property-description">
                        <p className="text-lg font-bold">{property.propertyTitle}</p>
                        <p className="text-sm text-gray-500">{property.address}</p>
                        <hr className="my-3" />
                        <Link to={`/property/${property._id}`} className="flex justify-end">
                          <Button buttonType={"primary"}>Book</Button>
                        </Link>
                      </div>
                    </Link>
                  )}
                </div>
              }
            </div>
          </div>

        </div>

        <div className="add-your-space-section flex flex-col items-center text-center">
          <h1 className="text-white text-3xl font-bold leading-tight mb-5">
            List Your Own Space
          </h1>
          <p className="text-gray-300 mb-10">
            More people use sharedspace.com to discover coworking spaces than <br />
            all the other platforms combined.
          </p>

          <Button buttonType={"primary"} onClick={onListYourSpace}>Start Now</Button>

        </div>

        <Footer />
      </div>
    </>
  );
}

export default HomePage