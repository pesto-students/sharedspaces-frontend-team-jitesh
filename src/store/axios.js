import axios from "axios";

const Axios = axios.create({
  baseURL: "http://localhost:5000/api",
  // baseURL:
  //   process.env.REACT_APP_BASE_URL,
});

export default Axios;
