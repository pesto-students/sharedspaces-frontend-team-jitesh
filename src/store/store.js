import { createStore, applyMiddleware } from "redux";
import thunk from "redux-thunk";
import { composeWithDevTools } from "redux-devtools-extension";
import { combineReducers } from "redux";
import siteReducer from "./reducers/siteReducer";

let middleware = [thunk];

const rootReducer = combineReducers({
  site: siteReducer,
});

const store = createStore(
  rootReducer,
  composeWithDevTools(applyMiddleware(...middleware))
);

export default store;
