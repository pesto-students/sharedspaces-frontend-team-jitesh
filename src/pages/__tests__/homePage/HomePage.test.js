import { render, screen, cleanup } from "@testing-library/react";
import { Provider } from "react-redux";
import { BrowserRouter } from 'react-router-dom';
import store from '../../../store/store'
import renderer from 'react-test-renderer'
import HomePage from '../../homePage/HomePage'

afterEach(() => {
  cleanup()
})


test('test matches snapshot', () => {
  const component = renderer.create(
    <BrowserRouter>
      <Provider store={store}>
        <HomePage />
      </Provider>
    </BrowserRouter>
  )

  let tree = component.toJSON()
  expect(tree).toMatchSnapshot();
})
