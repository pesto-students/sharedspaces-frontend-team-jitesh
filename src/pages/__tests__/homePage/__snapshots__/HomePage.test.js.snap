// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test matches snapshot 1`] = `
<div
  className="homepage-wrapper"
>
  <div
    className="search-wrapper"
  >
    <div
      className="search-bg flex"
    >
      <div
        className="left-bg bg-black"
      />
      <div
        className="right-bg bg-red-500"
      />
    </div>
    <div
      className="container-wrapper"
    >
      <div
        className="section fade-in-bottom"
      >
        <div
          className="left-section"
        >
          <div
            className="search-with-heading"
          >
            <h1
              className="text-white text-5xl md:text-3x1 font-bold py-10 leading-tight"
            >
              The  Best Place 
              <br />
              To Find Your
              <br />
              Office Space
            </h1>
            <form
              onSubmit={[Function]}
            >
              <div
                className="search-input-wrapper shadow-new bg-white rounded flex justify-between"
              >
                <input
                  name="search"
                  onChange={[Function]}
                  placeholder="Start your search..."
                  required={true}
                  type="text"
                  value={null}
                />
                <a
                  href="/search"
                  onClick={[Function]}
                >
                  <button
                    className="
            btn-wrapper
        bg-red-500 text-white border-2 border-red-500 hover:bg-red-500 hover:shadow-sm active:bg-red-800 active:shadow-sm active:text-white
        false
        false
        false
        undefined
        px-6 py-1.5 font-medium rounded shadow-md transition duration-150 ease-in-out  whitespace-nowrap cursor-pointer"
                    data-testid="button"
                    type="submit"
                  >
                    Search
                  </button>
                </a>
              </div>
            </form>
          </div>
        </div>
        <div
          className="right-section"
        >
          <img
            alt=""
            src="/assets/images/search-section.png"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="black-background"
  >
    <div
      className="property-list-wrapper"
    >
      <div
        className="popular-properties-section"
      >
        <h1
          className="text-black text-2xl font-bold py-10 leading-tight  "
        >
          Popular Properties
        </h1>
        <div
          className="property-list"
        />
      </div>
    </div>
  </div>
  <div
    className="add-your-space-section flex flex-col items-center text-center"
  >
    <h1
      className="text-white text-3xl font-bold leading-tight mb-5"
    >
      List Your Own Space
    </h1>
    <p
      className="text-gray-300 mb-10"
    >
      More people use sharedspace.com to discover coworking spaces than 
      <br />
      all the other platforms combined.
    </p>
    <button
      className="
            btn-wrapper
        bg-red-500 text-white border-2 border-red-500 hover:bg-red-500 hover:shadow-sm active:bg-red-800 active:shadow-sm active:text-white
        false
        false
        false
        undefined
        px-6 py-1.5 font-medium rounded shadow-md transition duration-150 ease-in-out  whitespace-nowrap cursor-pointer"
      data-testid="button"
      onClick={[Function]}
    >
      Start Now
    </button>
  </div>
  <div
    className="footer text-center"
    data-testid="footer"
  >
    <div
      className="footer-with-menu border-t-2 border-gray-700 py-10"
    >
      <div
        className="left"
      >
        <img
          alt=""
          className="logo"
          src="/assets/images/logo.png"
        />
      </div>
      <div
        className="right"
      >
        <nav
          className="right-menu"
        >
          <a
            className="rounded-lg pl-10 py-2 text-slate-700 font-medium hover:text-slate-100 transition duration-150 ease-in-out "
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
          <a
            className="rounded-lg pl-10 py-2 text-slate-700 font-medium hover:text-slate-100 transition duration-150 ease-in-out "
            href="/search"
            onClick={[Function]}
          >
            Explore
          </a>
          <a
            className="rounded-lg pl-10 py-2 text-slate-700 font-medium hover:text-slate-100 transition duration-150 ease-in-out "
            href="/"
            onClick={[Function]}
          >
            About
          </a>
          <a
            className="rounded-lg pl-10 py-2 text-slate-700 font-medium hover:text-slate-100 transition duration-150 ease-in-out "
            href="/"
            onClick={[Function]}
          >
            Contact
          </a>
        </nav>
      </div>
    </div>
    <p
      className="text-gray-700"
    >
      SharedSpace | copyright Â© 2022
    </p>
  </div>
</div>
`;
